orchagent/main.cpp

```diff
@@ -140,6 +145,121 @@ void init_gearbox_phys(DBConnector *applDb)
     delete tmpGearboxTable;
 }
 
+bool getSystemPortConfigList(vector<sai_system_port_config_t> &sysportcfglist)
+{
+    DBConnector cfgDb("CONFIG_DB", 0);
+    DBConnector appDb("APPL_DB", 0);
+
+    Table cfgDeviceMetaDataTable(&cfgDb, CFG_DEVICE_METADATA_TABLE_NAME);
+    Table cfgSystemPortTable(&cfgDb, CFG_SYSTEM_PORT_TABLE_NAME);
+    Table appSystemPortTable(&appDb, APP_SYSTEM_PORT_TABLE_NAME);
+
+    //At this point of time (i.e, at the time of orchagent start), the CONFIG_DB is completely fully
+    //populated with the contents of config_db.json.
+
+    string value;
+    //Get the swith type.
+    if(!cfgDeviceMetaDataTable.hget("localhost", "switch_type", value))
+    {
+        //Switch type is not configured. Consider it default = "switch" (regular switch)
+        value = "switch";
+    }
+
+    if(value != "voq" && value != "fabric" && value != "switch")
+    {
+        cout << "Invalid switch type " << value.c_str() << " configured";
+        return false;
+    }
+    gMySwitchType = value;
+
+    if(gMySwitchType != "voq")
+    {
+        //Non VOQ switch. Nothing to read
+        return true;
+    }
+
+    if(!cfgDeviceMetaDataTable.hget("localhost", "switch_id", value))
+    {
+        //VOQ switch id is not configured.
+        cout << "VOQ switch id is not configured";
+        return false;
+    }
+
+    if(value.size())
+        gVoqMySwitchId = stoi(value);
+
+    if(gVoqMySwitchId < 0)
+    {
+        cout << "Invalid VOQ switch id " << gVoqMySwitchId << " configured";
+        return false;
+    }
+
+    if(!cfgDeviceMetaDataTable.hget("localhost", "max_cores", value))
+    {
+        //VOQ max cores is not configured.
+        cout << "VOQ max cores is not configured";
+        return false;
+    }
+
+    if(value.size())
+        gVoqMaxCores = stoi(value);
+
+    if(gVoqMaxCores == 0)
+    {
+        cout << "Invalid VOQ max cores " << gVoqMaxCores << " configured";
+        return false;
+    }
+
+    vector<string> spKeys;
+    cfgSystemPortTable.getKeys(spKeys);
+
+    //Retrieve system port configurations
+    vector<FieldValueTuple> spFv;
+    sai_system_port_config_t sysport;
+    for ( auto &k : spKeys )
```

Comment:
Fix the format:

for (auto &k : spKeys)
{
}

Reply:
We'll fix this.

Reply:
Fixed